Description:  Hack broken rust target selection so it produces the right target on raspbian.
Author: Peter Michael Green <plugwash@raspbian.org>
diff --git a/build/moz.configure/rust.configure b/build/moz.configure/rust.configure
index 40f017da67..8dbbace178 100644
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -236,8 +236,13 @@ def rust_compiler(rustc_info, cargo_info):
 
 
 @depends(rustc, when=rust_compiler)
+@imports("sys")
+@imports(_from="__builtin__", _import='repr')
 @imports(_from="__builtin__", _import="ValueError")
 def rust_supported_targets(rustc):
+    #HACK: if there are multiple matches the algorithm uses the last one
+    #put arm-unknown-linux-gnuebihf on the end of the list so it gets preffered
+    out.append("arm-unknown-linux-gnueabihf")
     out = check_cmd_output(rustc, "--print", "target-list").splitlines()
     data = {}
     for t in out:
